lib/matplotlib/cm.py

```diff
@@ -701,6 +701,6 @@ def _ensure_cmap(cmap):
     """
     if isinstance(cmap, colors.Colormap):
         return cmap
-    return mpl.colormaps[
-        cmap if cmap is not None else mpl.rcParams['image.cmap']
-    ]
+    cmap_name = cmap if cmap is not None else mpl.rcParams["image.cmap"]
+    _api.check_in_list(sorted(_colormaps), cmap=cmap_name)
```

Comment:
Do you want to put a comment in saying you added this check for error raising type stability so we don't remove it later? Does it need to be sorted or can it be `list(_colormaps)`?

Reply:
The input is used in formatting the error to tell the user what the allowed values are.  Given the number of color maps I think sorting is a usability win / kindness for the user to discover what the correct spelling of the color map they tried to ask for is.

Reply:
I pushed a comment about why we are doing extra work.